### PR TITLE
Method invocation analyzer

### LINKED ISSUE #37: Potential Analyzer to complement USP0008 and make code more robust.
After looking at the behavior of the suppressor [USP0008: Don't flag private methods used with Invoke/InvokeRepeating or StartCoroutine/StopCoroutine as unused](https://github.com/microsoft/Microsoft.Unity.Analyzers/blob/master/doc/USP0008.md), it seems like it might be beneficial to provide an analyzer that will refactor the code.

A proposed solution would be to take the string literal parameter and look for a matching existing method name (allow for a case insensitive search to potentially correct an error in the string representation of the method name) and replace the string literal with the `nameof()` operator.  It has the further benefit of allowing the method to use a rename refactoring without remembering to update the string literals.

Resulting code would look something like the following:

```
using UnityEngine;

class Camera : MonoBehaviour
{
	void Start()
	{
		Invoke(nameof(InvokeMe), 10.0f);
	}

	private void InvokeMe()
	{
		// Invoke me
	}
}
```

### PR DESCRIPTION
**Pull request by sailro**
Fixes #37 and #65

#### Checklist
- [x] I have read the [Contribution Guide](/CONTRIBUTING.md) ;
- [x] There is an approved issue describing the change when contributing a new analyzer or suppressor ;
- [x] I have added tests that prove my fix is effective or that my feature works ;
- [x] I have added necessary documentation (if appropriate) ;

#### Short description of what this resolves:
Using `Invoke`, `InvokeRepeating`, `StartCoroutine` or `StopCoroutine` with a first argument being a string literal is not type safe. Instead it's recommanded to use the `nameof` operator or a direct call for coroutines. The further benefit of doing this is the ability for the method to use a rename refactoring without remembering to update the string literals.

#### Changes proposed in this pull request:
Added one diagnostic and two code fixes.

Note that `ConsistencyTests` will fail because of the analyzer ID. That's expected while all other PRs are not yet merged.

### CODE REVIEW

File: doc/UNT0016.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+```csharp
+using UnityEngine;
+using System.Collections;
+
+class Camera : MonoBehaviour
+{
+    void Start()
+    {
+	    Invoke("InvokeMe", 10.0f)
```
Comment (jbevain) [inline doc/UNT0016.md:15]: Whitespaces are looking off.

File: doc/UNT0016.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+```csharp
+using UnityEngine;
+using System.Collections;
+
+class Camera : MonoBehaviour
+{
+    void Start()
+    {
+	    Invoke(nameof(InvokeMe), 10.0f)
```
Comment (jbevain) [inline doc/UNT0016.md:43]: Same.

**Comment by jbevain** [comment]
This is looking good. One thing though, we need tests for the case where the method is called not on `this` or `base` and defined in another type, like:

```csharp
class A : MonoBehaviour
{
    private B b;

    void Update()
    {
        b.InvokeRepeating("Foo");
    }
}

class B : MonoBehaviour
{
    void Foo()
    {
    }
}
```

**Comment by sailro** [comment]
> This is looking good. One thing though, we need tests for the case where the method is called not on `this` or `base` and defined in another type, like:
> 
> ```cs
> class A : MonoBehaviour
> {
>     private B b;
> 
>     void Update()
>     {
>         b.InvokeRepeating("Foo");
>     }
> }
> 
> class B : MonoBehaviour
> {
>     void Foo()
>     {
>     }
> }
> ```

We can properly detect this pattern for suppressing IDE0051 for `Foo`, but we cannot offer `nameof(A.B.Foo)` or `nameof(B.Foo)` instead of `"Foo"` because `Foo is inaccessible due to its protection level`.

**Comment by jbevain** [comment]
Exactly, we want to test we're not offering a codefix that would break the compilation.

**Comment by jbevain** [comment]
Thanks!

**Pull request merged**